blocks/datasets/streams.py

```diff
@@ -314,6 +314,40 @@ def _cache(self):
             cache.extend(data)
 
 
+class SortMapping(object):
+    """Callable class for creating sort mappings without nesting functions."""
+    def __init__(self, key, reverse):
+        self.key = key
+        self.reverse = reverse
+
+    def __call__(self, x):
+        indices = [i for (v, i) in
+                   sorted(((v, i) for (i, v) in enumerate(x[0])),
```

Comment:
Ah, this line is wrong. I have to rethink this and write a test that has an example with multiple sources because I'm sure this version wouldn't pass that.